kafka_topic_dumper/kafka_client.py

```diff
@@ -185,73 +208,146 @@ def get_messages(self, num_messages_to_consume, max_package_size_in_msgs,
+                                file_name=file_name)
+            self._send_dump_file(dir_path=dir_path, file_name=file_name,
+                                 bucket_name=bucket_name, s3_client=s3_client)
             remaining_messages -= batch_size
             num_dumped_messages += batch_size
 
         logger.info('Dump done!')
 
-        self._close_consumer()
-
-    def reload_kafka_server(self, bucket_name, dir_path, dump_prefix=None):
-        self._get_producer()
-
-        # get file list
-        s3_client = boto3.client('s3')
+    def _get_file_names(self, bucket_name, dump_prefix, s3_client):
         paginator = s3_client.get_paginator('list_objects_v2')
 
         if dump_prefix is None:
+            prefix = self.s3_path
+            sufix = '-'
+            reg = re.compile('^{}(.*){}$'.format(prefix, sufix))
+
             response_iterator = paginator.paginate(Bucket=bucket_name,
+                                                   Prefix=prefix,
                                                    Delimiter='-')
+            def striper(string):
+                match = reg.match(string['Prefix'])
+                if match is not None:
+                    return match.group(1)
+
             prefixes = []
             for response in response_iterator:
-                response_prefixes = [p['Prefix'].replace('-', '')
-                                     for p in response['CommonPrefixes']]
+                response_prefixes = filter(
+                    lambda x: x is not None,
+                    map(striper, response['CommonPrefixes']))
                 prefixes += response_prefixes
+
             dump_prefix = max(prefixes)
             logger.info('Prefix chosen was <{}>'.format(dump_prefix))
 
+        dump_path = prefix + dump_prefix
+
         response_iterator = paginator.paginate(Bucket=bucket_name,
-                                               Prefix=dump_prefix)
+                                               Prefix=dump_path)
         file_names = []
         for response in response_iterator:
             if response['KeyCount'] > 0:
                 file_names = [(f['Key'], f['Size'])
                               for f in response['Contents']]
         file_names.sort()
+        return dump_prefix, file_names
+
+    def _set_test_topic(self, test_id):
+        _, _, end_offsets = self._get_offsets()
+
+        if not end_offsets:
+            msg = 'Can not find offsets for topic <{}>'
+            raise Exception(msg.format(self.topic))
+
+        test_offsets = {}
+
+        for partition, offset in end_offsets.items():
+            test_offsets[partition.partition] = offset
+
+        test_state = {
+            'test_id': test_id,
+            'topic_name': self.topic,
+            'offsets': test_offsets}
+
+        future = self.producer.send(
+            topic=self.dump_state_topic,
+            key=test_id,
+            value=json.dumps(test_state))
+        future.get(timeout=self.timeout)
```

Comment:
Isn't this timeout too low?

Reply:
Not really, we are using this inside AWS. so it should not take so long.